src/ImageView.jl

```diff
@@ -148,21 +151,41 @@ end
 function imshow!(canvas::GtkObservables.Canvas,
                  img::AbstractMatrix,
                  annotations::Annotations=annotations())
+    pyr = ImagePyramid(img)
     draw(canvas, annotations) do cnvs, anns
-        copy_with_restrict!(cnvs, img)
+        copy_with_restrict!(cnvs, pyr)
         set_coordinates(cnvs, axes(img))
         draw_annotations(cnvs, anns)
     end
     nothing
 end
 
-function copy_with_restrict!(cnvs, img::AbstractMatrix)
-    imgsz = size(img)
+# cache of the image plus downscaled versions
+ImagePyramid(img::AbstractMatrix) = Any[img]
+
+function ImagePyramid(img::Observable)
+    pyr = Any[img[]]
+    on(img) do image
+        push!(empty!(pyr),image)
+    end
+    pyr
+end
+
+function get_image(p, i)
```

Comment:
```suggestion
function get_restricted_image!(p, i)
```

Can we use more descriptive variable names than `p` and `i`?